### PR TITLE
this brings python & ruby options code and tests into alignment

### PR DESCRIPTION
**Pull request by titusfortner**
I updated the Ruby code with some of the features implemented in Python, as well as applicable tests. 

Mostly this is a big update to the Python code ("look ma, I wrote real Python code")
It might not be Pythonic? It might not be what we want to do? It probably has PEP issues?

Essentially, I metaprogrammed the `__getattr__` and `__setattr__` similar to how I would do it in Ruby, rather than creating a million `@property`. The advantage is all of the values can be directly accessed; hopefully this is good for Python?

I'm a little concerned with inconsistency between camel case and snake case. I think latest PEP says snake for everything except class name? Or is that just people on SO?

I removed the default test name, since we want Sauce to default to the "Untitled blah blah blah" rather than having all the tests with the same generic name.

I removed the ability for Options class to take ChromeOptions, etc, since we should be implementing that in Session. (at least that's what I was arguing for last week, we can reconsider if we don't like the code).

Let me know what you think of my Python and if it is doing anything especially weird or has some extra duplication. Looking forward to feedback on the approach.

### CODE REVIEW

File: ruby/spec/options_spec.rb [hunk 1]
```diff
@@ -53,31 +57,51 @@ module SimpleSauce
...
+
+        expect(sauce_opts.capabilities['sauce:options']).to eq('idleTimeout' => 3,
+                                                               'recordScreenshots' => false,
+                                                               'name' => 'TEST NAME',
+                                                               'build' => 'CUSTOM BUILD')
+      end
     end
   end
-end
+end
```
Comment (nadvolod) [inline ruby/spec/options_spec.rb]: Codacy Issue found: [Final newline missing.](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4675472)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity decreasing per file
==============================
+ python/simplesauce/options.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4675472&bid=15644313)

File: ruby/spec/options_spec.rb (hunk 1 above)
Comment (titusfortner) [reply ruby/spec/options_spec.rb]: oops

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ python/simplesauce/options.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4675472&bid=15644313)

**Comment by nadvolod** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ python/simplesauce/options.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4675472&bid=15644313)

File: python/simplesauce/options.py [hunk 2]
```diff
@@ -1,117 +1,139 @@
...
+            job_name = re.search("[^/]+$", os.environ.get("TRAVIS_REPO_SLUG"))
+            self.build = "{}: {}".format(job_name, os.environ['TRAVIS_JOB_NUMBER'])
+        # Circle
+        elif os.environ.get('CIRCLE_JOB'):
+            self.build = "{}: {}".format(os.environ['CIRCLE_JOB'], os.environ['CIRCLE_BUILD_NUM'])
+        # Gitlab
+        elif os.environ.get('CI'):
+            self.build = "{}: {}".format(os.environ['CI_JOB_NAME'], os.environ['CI_JOB_ID'])
+        else:
+            self.build = 'Build Time: {}'.format(datetime.utcnow())
```
Comment (joshmgrant) [inline python/simplesauce/options.py:74]: Love it! I tried to take the lazy way out but this is better.

**Comment by titusfortner** [comment]
This needs to go in a different direction, so I'm closing it. I merged in the build name updates to master since we seem to agree that's the right direction. I'll PR something new soon.

**Pull request closed without merging**